test/unit/kubernetes-deploy/kubernetes_resource/pod_test.rb

```diff
@@ -170,6 +170,21 @@ def test_deploy_failed_is_false_for_evicted
     assert_nil pod.failure_message
   end
 
+  def test_deploy_failed_is_false_for_preempting
+    container_state = pod_spec.merge(
+      "status" => {
```

Comment:
Are you confident that this is what the status will look like (or at least the reason). Otherwise this PR wont work.

Reply:
Looks like it gets set here 👍 

https://github.com/kubernetes/kubernetes/blob/817869e78af9d3250a36fce845336535dd29d0e4/pkg/kubelet/preemption/preemption.go#L106

Reply:
Thanks both!